wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/interfaces/EntityIdValue.java

```diff
@@ -107,4 +107,15 @@
 	 */
 	String getSiteIri();
 
+	/**
+	 * Checks whether this entity ID is valid. Invalid IDs are commonly used as
+	 * placeholders when creating new entitites.
+	 * 
+	 * @return {@code true} if this is a valid ID, {@code false} for placeholder or
+	 *         otherwise invalid ID
+	 */
+	default boolean isValid() {
+		return !getSiteIri().equals(SITE_LOCAL);
+	}
```

Comment:
`EntityUpdateBuilder` classes really need `isValid()` method. If `SITE_LOCAL` IDs are generally valid and only null IDs are invalid, I can change the `isValid()` implementation accordingly. It is always possible to override it if someone defines new ID types.

Reply:
Yes I understand they need this method - my comment is simply about the *name* of that method. I think this name should be changed and the corresponding error message thrown by the constructors.

Reply:
So what should the method name be? Naming it `isLocal()` sounds wrong, because `EntityUpdateBuilder` classes don't really care whether the ID is local or not. They just need to know when it is a real ID and when it is not. Perhaps `addForm` and `addSense` could be changed to always discard the ID whether it is already a null ID or not, so they would not need `isValid()` method. Validation checks however do need some method on `EntityIdValue` to detect the null placeholder IDs. Perhaps it could be named `isNull()` then? Or `isPlaceholder()`? Messages can be updated, although they should not pretend that builders have knowledge about what's stored in remote Wikibase instance and what's not.

Reply:
Well for me that's what this SITE_LOCAL siteIRI is about - representing entities which are only known locally. I would not call it `isNull` because it would be too similar to things like `ItemIdValue.NULL`. Why not `isPlaceholder`! Concerning the messages, I just think it's vague to say that something is "invalid"… it does not really help the user understand what is wrong :)

Reply:
I am going to modify the logic of `isValid()` to only return `false` when the ID is `ItemIdValue.NULL` (or one of the other such predefined values). Change of method name is an addition to that, not an alternative. I can use name `isPlaceholder()` instead of `isValid()`. It's arguably more descriptive.

Reply:
I have replaced `isValid()` with `isPlaceholder()`. It returns `true` only for zero IDs, which are used as placeholders in WDTK.